plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/DialogStagingDirectoryProvider.java

```diff
@@ -0,0 +1,27 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import org.eclipse.swt.widgets.DirectoryDialog;
+import org.eclipse.swt.widgets.Shell;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.Messages;
+
+/**
+ * Prompts the user to select a directory for staging.
+ */
+// TODO should be moved to separate UI bundle (if kept at all in deploy's final implementation)
+public class DialogStagingDirectoryProvider implements StagingDirectoryProvider {
+
+  private Shell parentShell;
+
+  public DialogStagingDirectoryProvider(Shell shell) {
+    this.parentShell = shell;
+  }
+
+  @Override
+  public String getStagingDirectory() {
+    DirectoryDialog directoryDialog = new DirectoryDialog(parentShell);
```

Comment:
We should probably use `DirectoryDialog(shell, SWT.APPLICATION_MODAL | SWT.SHEET)` to get nice appearance on OS X and Unity.